wfe2/wfe.go

```diff
@@ -1548,7 +1549,23 @@ func (wfe *WebFrontEndImpl) Certificate(ctx context.Context, logEvent *web.Reque
 		requesterAccount = acct
 	}
 
+	requestedChain := 0
 	serial := request.URL.Path
+
+	// An alternate chain may be requested with the request path {serial}/{chain}, where chain
+	// is a number - an index into the slice of chains for the issuer. If a specific chain is
+	// not requested, then it defaults to zero - the default certificate chain for the issuer.
+	serialAndChain := strings.SplitN(serial, "/", 2)
+	if len(serialAndChain) == 2 {
+		idx, err := strconv.Atoi(serialAndChain[1])
+		if err != nil {
+			wfe.sendError(response, logEvent, probs.NotFound("Certificate not found"), nil)
+			return
+		}
+		serial = serialAndChain[0]
+		requestedChain = idx
```

Comment:
Personally I would just move the negative check you are doing on L1653 here since it's always going to be an invalid request and save some DB resources.

Reply:
It felt a bit unsafe to put it earlier because there's ~100 LOC between L1653 and the actual site where a panic could occur. Would you be OK with checking negative at both sites?

Reply:
As long as it's not being manipulated between here and there it should be fine.